modules/xfem/src/base/XFEM.C

```diff
@@ -882,6 +929,148 @@ XFEM::initCutIntersectionEdge(
   return does_intersect;
 }
 
+bool
+XFEM::healMesh()
+{
+  bool mesh_changed = false;
+
+  std::set<Node *> nodes_to_delete;
+  std::set<Node *> nodes_to_delete_displaced;
+  std::set<unsigned int> cutelems_to_delete;
+
+  for (unsigned int i = 0; i < _geometric_cuts.size(); ++i)
+  {
+    if (_geometric_cuts[i]->healMesh())
+    {
+      for (ElementPairLocator::ElementPairList::iterator it =
```

Comment:
A range-based for loop would really clean this up!

Reply:
will do :-)